presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -167,6 +167,58 @@ public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
         }
     }
 
+    @Test
+    public void testCreateWithFailureDoesntLeaveBehindOrphanedTable()
+            throws Exception
+    {
+        String schemaName = format("tmp_schema_%s", UUID.randomUUID().toString().replaceAll("-", ""));
+        try (AutoCloseable schema = withSchema(schemaName);
+                AutoCloseable table = withTable(format("%s.test_table_cleanup", schemaName), "(x INTEGER)");
+                AutoCloseable type = withType(format("%s.test_table_cleanup_type", schemaName), "(x INTEGER)")) {
+            assertQueryFails(format("CREATE TABLE %s.test_table_cleanup (column_a DATE)", schemaName), format(".*'postgresql.%s.test_table_cleanup' already exists", schemaName));
+            assertQueryFails(format("CREATE TABLE %s.test_table_cleanup_type (column_a DATE)", schemaName), ".*\"test_table_cleanup_type\" already exists.*");
+            assertQuery(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", schemaName), "VALUES 'test_table_cleanup'");
+        }
+    }
+
+    @Test
+    public void testFailsWhenNotSpecifiedColumnIsNotNull()
+            throws SQLException
+    {
+        try {
+            String catalog = getSession().getCatalog().get();
+            assertUpdate("CREATE TABLE test_not_null_with_unspecified (" +
+                    "column_a DATE," +
+                    "column_b DATE NOT NULL" +
+                    ")");
+
+            assertQueryFails("INSERT INTO tpch.test_not_null_with_unspecified(column_a) VALUES (date '2012-12-31')",
+                    "(?s).*ERROR: null value in column \"column_b\" violates not-null constraint.*");
+        }
+        finally {
+            assertUpdate("DROP TABLE tpch.test_not_null_with_unspecified");
+        }
+    }
+
+    @Test
+    public void testFailsWhenInsertNullIntoNotNullColumn()
```

Comment:
Let's merge these two tests together and rename the test to `testInsertIntoNotNullColumn`